features/conversation-list/hooks/use-delete-dm.ts

```diff
@@ -0,0 +1,116 @@
+import { showActionSheetWithOptions } from "@/components/StateHandlers/ActionSheetStateHandler";
+import { useCurrentAccount } from "@/data/store/accountsStore";
+import { usePreferredInboxName } from "@/hooks/usePreferredInboxName";
+import { translate } from "@/i18n";
+import {
+  getConversationDataQueryData,
+  setConversationDataQueryData,
+} from "@/queries/use-conversation-data-query";
+import { useDmPeerInboxId } from "@/queries/useDmPeerInbox";
+import { actionSheetColors } from "@/styles/colors";
+import { useAppTheme } from "@/theme/useAppTheme";
+import { deleteTopic } from "@/utils/api/topics";
+import { captureErrorWithToast } from "@/utils/capture-error";
+import { DmWithCodecsType } from "@/utils/xmtpRN/client.types";
+import { consentToInboxIdsOnProtocolByAccount } from "@/utils/xmtpRN/contacts";
+import { useMutation } from "@tanstack/react-query";
+import { useCallback } from "react";
+
+export const useDeleteDm = (dm: DmWithCodecsType) => {
+  const { theme } = useAppTheme();
+  const colorScheme = theme.isDark ? "dark" : "light";
+
+  const currentAccount = useCurrentAccount()!;
+
+  const { data: peerInboxId } = useDmPeerInboxId({
+    account: currentAccount,
+    topic: dm.topic,
+  });
+
+  const preferredName = usePreferredInboxName(peerInboxId);
+
+  const { mutateAsync: deleteDmAsync } = useMutation({
+    mutationFn: () =>
+      deleteTopic({
+        account: currentAccount,
+        topic: dm.topic,
+      }),
+    onMutate: () => {
+      const previousIsDeleted = getConversationDataQueryData({
+        account: currentAccount,
+        topic: dm.topic,
+        context: "useDeleteDm",
+      })?.isDeleted;
+
+      setConversationDataQueryData({
+        account: currentAccount,
+        topic: dm.topic,
+        context: "useDeleteDm",
+        data: { isDeleted: true },
+      });
+
+      return { previousIsDeleted };
+    },
+    onError: (error, _, context) => {
+      setConversationDataQueryData({
+        account: currentAccount,
+        topic: dm.topic,
+        context: "useDeleteDm",
+        data: { isDeleted: context?.previousIsDeleted },
+      });
+    },
+  });
+
+  return useCallback(() => {
```

Comment:
I think this should be up to the caller:

Something like

`useDeleteDm({onDeleteTriggered: () => void})` and then a `useDeleteDmWithActionSheet` but if this is the only usage than that's okay. It's just weird having these ui actions in the hook